pinot-broker/src/main/java/com/linkedin/pinot/broker/routing/builder/BasePartitionAwareRoutingTableBuilder.java

```diff
@@ -61,17 +61,15 @@
 
   protected ZkHelixPropertyStore<ZNRecord> _propertyStore;
   protected SegmentZKMetadataPrunerService _pruner;
-  protected TableConfig _tableConfig;
   protected Random _random = new Random();
-  protected int _numReplicas;
+  protected volatile int _numReplicas;
```

Comment:
Curious, why does this need to be volatile?

Reply:
As discussed offline, write to volatile int is atomic.